### DISCUSSION TITLE
Removing script from a room

### DISCUSSION THREAD

**Discussion by mcarella** [evennia/evennia#2480] [Evennia Questions] [answered]
Hello everyone if I can associate a script to a room like this:

`@script here = typeclasses.scripts.Weather`

Is there a similar command to remove it?

 I'm thinking about something

`@script here = None`

Thank you in advance :-)

**Comment by kovitikus** [comment, marked as answer]
`script/stop here = typeclasses.scripts.Weather` will obviously stop it. Will it remove it from the room? I don't know. The room is associated with the script via a script property.

Well, here, I found an answer for you. 

https://www.evennia.com/docs/latest/Scripts.html#defining-new-scripts

Further down on this page you can see a snippet about removing a script from a room. This is done via Python code, but you can execute that in your game as well.

`py self.location.scripts.delete('weather_script')`

As long as your character is standing in the room in question, of course.

`weather_script` is the name of the script and if you need to find the name of YOUR script, you just use the scripts command.

![scripts command output](https://i.imgur.com/8wsWav8.png)

Good luck, hope this helps.

**Reply by mcarella** [reply]
It works! Thanks a lot!

**Reply by pruuk** [reply]
One thing to note:
Using .delete on the scripts sometimes does not trigger the at_stop() method on the script as expected. I ran into this with my combat handler script that does twitch-style combat. I had to use .stop()

**Reply by kovitikus** [reply]
Yeah `.delete()` isn't a method on the Script typeclass itself, it's something done much higher up in Django if I remember correctly.

Right now, the best approach is to `.stop()`, which should also automatically call `.delete()` from higher up and clean up the script for you.

In a future version, @Griatch has refactored scripts so that stop no longer automatically deletes, and you will have to manually call delete.

**Reply by pruuk** [reply]
I can just call .delete() at the end of at_stop() though, correct? Then I get the same effect of running my cleanups?

**Reply by kovitikus** [reply]
I don't know the exact flow here, you'll have to wait for @Griatch to chime in.

**Reply by Griatch** [reply]
With the latest `develop` you should be able to call `.delete` at the end of `at_stop` - for `master` branch scripts you are probably best off to just call `stop` - this should delete the script as well.
